test/validationRules/required-fields.js

```diff
@@ -31,14 +32,42 @@ const requiredFieldsTestCases = {
     },
     {
       code:
-        "const x = gql`query { greetings { hello ... on Greetings { foo } } }`",
+        "const x = gql`query { greetings { hello ... on Greetings { hello } } }`",
       errors: [
         {
           message: `'id' field required on 'greetings'`,
           type: "TaggedTemplateExpression"
         }
       ]
-    }
+    },
+    {
+      code: 'const x = gql`query { greetings { hello ...GreetingsFragment} }`',
+      errors: [
+        {
+          message: `'id' field required on 'greetings'`,
+          type: 'TaggedTemplateExpression',
+        },
+      ],
+    },
+    {
+      code: 'const x = gql`fragment Name on Greetings { hello }`',
+      errors: [
+        {
+          message: `'id' field required on 'fragment Name on Greetings'`,
+          type: 'TaggedTemplateExpression',
+        },
+      ],
+    },
+    {
+      code:
+        "const x = gql`query { greetingOrStory { ... on Greetings { id } ... on Story { comments { text } } } }`",
+      errors: [
+        {
+          message: `'id' field required on '... on Story'`,
+          type: "TaggedTemplateExpression"
+        }
+      ]
+    },
```

Comment:
Also worth testing an interface where `id` is a shared field?

Reply:
Done!